contracts/hooks/axelar/src/lib.rs

```diff
@@ -0,0 +1,292 @@
+    hpl_ownable::initialize(deps.storage, &owner)?;
+
+    let destination_chain = &msg.destination_chain;
+    let destination_contract = &msg.destination_contract;
+    let destination_ism = &msg.destination_ism;
+    let axelar_gateway_channel = &msg.axelar_gateway_channel;
+    let gas_token = &msg.gas_token;
+    let mailbox: Addr = deps.api.addr_validate(&msg.mailbox)?;
+
+    DESTINATION_CHAIN.save(deps.storage, destination_chain)?;
+    DESTINATION_CONTRACT.save(deps.storage, destination_contract)?;
+    DESTINATION_ISM.save(deps.storage, destination_ism)?;
+    AXELAR_GATEWAY_CHANNEL.save(deps.storage, axelar_gateway_channel)?;
+    GAS_TOKEN.save(deps.storage, gas_token)?;
+    MAILBOX.save(deps.storage, &mailbox)?;
+
+    Ok(Response::new().add_event(
+        new_event("initialize")
+            .add_attribute("sender", info.sender)
+            .add_attribute("owner", owner)
+            .add_attribute("destination_chain", destination_chain)
+            .add_attribute("destination_contract", destination_contract)
+            .add_attribute("destination_ism", destination_ism)
+            .add_attribute("axelar_gateway_channel", axelar_gateway_channel),
+    ))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        // TODO: maybe add SetWormholeCore Msg
+        ExecuteMsg::Ownable(msg) => Ok(hpl_ownable::handle(deps, env, info, msg)?),
+        ExecuteMsg::PostDispatch(msg) => post_dispatch(deps, env, info, msg),
+        ExecuteMsg::RegisterDestinationContract(msg) => {
+            register_destination_contract(deps, info, msg)
+        }
+    }
+}
+
+fn register_destination_contract(
+    deps: DepsMut,
+    info: MessageInfo,
+    msg: RegisterDestinationContractMsg,
+) -> Result<Response, ContractError> {
+    ensure_eq!(
+        get_owner(deps.storage)?,
+        info.sender,
+        ContractError::Unauthorized {}
+    );
+
+    let destination_contract = &msg.destination_contract;
+    DESTINATION_CONTRACT.save(deps.storage, destination_contract)?;
+
+    Ok(Response::new().add_event(
+        new_event("register_destination_contract")
+            .add_attribute("destination_contract", destination_contract),
+    ))
+}
+
+fn post_dispatch(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    req: PostDispatchMsg,
+) -> Result<Response, ContractError> {
+    // Ensure message_id matches latest dispatch from mailbox
+    let mailbox = MAILBOX.load(deps.storage)?;
+    let latest_dispatch_resp = deps
+        .querier
+        .query_wasm_smart::<LatestDispatchedIdResponse>(
+            &mailbox,
+            &MailboxQueryMsg::LatestDispatchId {}.wrap(),
+        )
+        .or_else(|_| return Err(ContractError::LastDispatchQueryFailed {}));
+
+    let latest_dispatch_id = latest_dispatch_resp.unwrap().message_id;
```

Comment:
unwrap() is unsafe. Can you do this query analogous to your wormhole query for latest_dispatch_id?